### PR TITLE
Feat/moving media

### LINKED ISSUE #452: [feat] Move functionality for medias
Right now, users have no way to move images or files between folders. To make the nested medias functionality complete, we should support media-moving, so that users will not need to go to Github to move their media.

Link to [Figma design file](https://www.figma.com/file/4qNhGTBMs521pDvCAIh4ON/IsomerCMS?node-id=1024%3A8585)

### PR DESCRIPTION
**Pull request by gweiying**
This PR introduces the feature to move media between folders. Previously in #441, we've refactored our Media layouts to support nested media folders, but we did not support file moving between folders. This closes #452. 
This PR should be reviewed with [#isomercms-backend/178](https://github.com/isomerpages/isomercms-backend/pull/178).

#### Other minor changes introduced in this PR:
- [feature] duplicate filename checking on `MediaSettingsModal` ** (see below)
- [refactor] removes call to retrieve media content in `MediaSettingsModal` on the frontend during rename operations
- [refactor] generalizes `FileMoveDropdownModal` to accept path query of any length
- [fix] bug in Media Breadcrumb

#### Technical Notes
Even though we introduce duplicate filename checking on `MediaSettingsModal`, this functionality will not work in `EditPage` and `FormFieldMedia` as `MediaSettingsModal` (where the check for duplication occurs) and `MediaModal` (where the call to retrieve media is performed) do not share state. We should tackle this issue in a later refactor. To simplify this, we can consider using hooks and/or context to abstract out the logic and state in `MediaSettingsModal` and `MediaModal`  for reuse. 

#### UX Notes
After consulting with Nat, we've decided on some of the following UX decisions:
- Remove the bounce hover state for `MediaCards` in favor of a pointer to standardize hover state across CMS 
- Preserve the `onClick -> Edit media` functionality of `MediaCards` in `Media`

### CODE REVIEW

**Review by alexanderleegs — CHANGES_REQUESTED** [review]
Feature looks great, just some minor nits

File: src/layouts/Media.jsx [hunk 1]
```diff
@@ -384,6 +476,29 @@ const Media = ({ match: { params: { siteName, customPath } }, location, mediaTyp
         />
         )
       }
+      {
+        isMoveModalActive && selectedMedia
+        && (
+          <GenericWarningModal
+            displayTitle="Warning"
+            displayText={`Moving ${mediaType.slice(0,-1)} to a different folder might lead to user confusion. You may wish to change the permalinks for this ${mediaType.slice(0,-1)} afterwards.`}
```
[the remark this answers is not in the document]
Comment (gweiying) [reply src/layouts/Media.jsx]: fixed in 1f3d419

File: src/components/media/MediaSettingsModal.jsx [hunk 2]
```diff
@@ -187,10 +127,10 @@ MediaSettingsModal.propTypes = {
     path: PropTypes.string,
     content: PropTypes.string,
   }).isRequired,
+  mediaFileNames: PropTypes.arrayOf(PropTypes.string),
   type: PropTypes.oneOf(['images', 'files']).isRequired,
   siteName: PropTypes.string.isRequired,
   onClose: PropTypes.func.isRequired,
-  onSave: PropTypes.func.isRequired,
```
[the remark this answers is not in the document]
Comment (gweiying) [reply src/components/media/MediaSettingsModal.jsx]: Good catch, must have removed it by accident. Thanks! cb35429

**Review by alexanderleegs — APPROVED** [review]
lgtm!

**Pull request merged**